test/boostTest.cpp

```diff
@@ -153,3 +153,18 @@ test_suite* init_unit_test_suite( int /*argc*/, char* /*argv*/[] )
 
 	return 0;
 }
+
+// BOOST_TEST_DYN_LINK should be defined if user want to link against shared boost test library
+#ifdef BOOST_TEST_DYN_LINK
+
+// Because we want to have customized initialization function and support shared boost libraries at the same time,
+// we are forced to customize the entry point.
+// see: https://www.boost.org/doc/libs/1_67_0/libs/test/doc/html/boost_test/adv_scenarios/shared_lib_customizations/init_func.html
+
+int main(int argc, char* argv[])
```

Comment:
Still no way to _always_ use the custom main function?

Reply:
It only states you have to use custom entry point for shared libs. But nothing prevents you from using it for static libs as well.

Reply:
We should probably look into that, but maybe we should create another issue for overhauling all these parts of the boost test setup (@axic already noted somewhere that we use deprecated features anyways) and queue that for 0.5.1 instead?

Reply:
Tracked in #4730 now.